crypto/fipsmodule/evp/evp_ctx_test.cc

```diff
@@ -0,0 +1,348 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include "internal.h"
+
+#include <gtest/gtest.h>
+#include <openssl/ec_key.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/kdf.h>
+#include <openssl/rsa.h>
+
+#include "../../internal.h"
+#include "internal.h"
+
+class EvpPkeyCtxCtrlStrTest : public ::testing::Test {
+ protected:
+  void SetUp() override {}
+
+  void TearDown() override {}
+};
+
+class EvpPkeyCtxCtrlStrParamTest : public testing::TestWithParam<const char *> {
+protected:
+  void SetUp() override {}
+
+  void TearDown() override {}
+};
+
+static bssl::UniquePtr<EVP_PKEY_CTX> gen_RSA() {
+  EVP_PKEY *raw = nullptr;
+  bssl::UniquePtr<EVP_PKEY_CTX> keygen_ctx(
+      EVP_PKEY_CTX_new_id(EVP_PKEY_RSA, nullptr));
+  if (!EVP_PKEY_keygen_init(keygen_ctx.get()) ||
+      !EVP_PKEY_CTX_set_rsa_keygen_bits(keygen_ctx.get(), 2048) ||
+      !EVP_PKEY_keygen(keygen_ctx.get(), &raw)) {
```

Comment:
surprisingly, we're seeing [a small leak here](https://us-west-2.codebuild.aws.amazon.com/project/eyJlbmNyeXB0ZWREYXRhIjoiWk5IUWJGRGxBcTJ2Mkp4WGF3dnBwYjc5V0ZZYSt5SVVGbkwvODkydTNTaVQ2V2FMN3hwa0tjSWNFemw2QWtCWW5welFWV3lpRFpKVitwejgvelhpRWh3NDNqcWhKalpPYW9hL2tLMDlJSDFPT1NkNyIsIml2UGFyYW1ldGVyU3BlYyI6Ilp4VXRNQXFGM1BZYVlaRkIiLCJtYXRlcmlhbFNldFNlcmlhbCI6MX0%3D/build/a6a2177f-6384-4404-b1fd-13b0aa186273):

```
==3352== 4,120 (32 direct, 4,088 indirect) bytes in 1 blocks are definitely lost in loss record 38 of 38
==3352==    at 0x6A3F86F: malloc (vg_replace_malloc.c:381)
==3352==    by 0xA53C3D: OPENSSL_malloc (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0xA53CB8: OPENSSL_zalloc (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0xB210F3: EVP_PKEY_new (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0xB23DBA: EVP_PKEY_keygen (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0x79D118: gen_RSA() (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0x79D197: EvpPkeyCtxCtrlStrTest_RsaMissingValue_Test::TestBody() (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0xA024F1: void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0x9FD378: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0x9DFCCD: testing::Test::Run() (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0x9E048C: testing::TestInfo::Run() (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
==3352==    by 0x9E0BF8: testing::TestSuite::Run() (in /codebuild/output/src2395812577/src/github.com/aws/aws-lc/test_build_dir/crypto/crypto_test)
```

should we wrap `raw` in a `bssl::UniquePtr`?